docs/griptape-framework/structures/tasks.md

```diff
@@ -667,3 +667,139 @@ pipeline.add_task(
+from griptape.tools import (
+    TaskMemoryClient,
+    WebScraper,
+    WebSearch,
+)
+
+researcher = Agent(
+    tools=[
+        WebSearch(
+            google_api_key=os.environ["GOOGLE_API_KEY"],
+            google_api_search_id=os.environ["GOOGLE_API_SEARCH_ID"],
+            off_prompt=False,
+        ),
+        WebScraper(
+            off_prompt=True,
+        ),
+        TaskMemoryClient(off_prompt=False),
+    ],
+    rulesets=[
+        Ruleset(
+            name="Position",
+            rules=[
+                Rule(
+                    value="Senior Research Analyst",
+                )
+            ],
+        ),
+        Ruleset(
+            name="Objective",
+            rules=[
+                Rule(
+                    value="Uncover cutting-edge developments in AI and data science",
+                )
+            ],
+        ),
+        Ruleset(
+            name="Background",
+            rules=[
+                Rule(
+                    value="""You work at a leading tech think tank.,
+                    Your expertise lies in identifying emerging trends.
+                    You have a knack for dissecting complex data and presenting actionable insights."""
+                )
+            ],
+        ),
+        Ruleset(
+            name="Desired Outcome",
+            rules=[
+                Rule(
+                    value="Full analysis report in bullet points",
+                )
+            ],
+        ),
+    ],
+)
+
+writer = Agent(
+    input_template="Instructions: {{args[0]}}\nContext: {{args[1]}}",
+    rulesets=[
+        Ruleset(
+            name="Position",
+            rules=[
+                Rule(
+                    value="Tech Content Strategist",
+                )
+            ],
+        ),
+        Ruleset(
+            name="Objective",
+            rules=[
+                Rule(
+                    value="Craft compelling content on tech advancements",
+                )
+            ],
+        ),
+        Ruleset(
+            name="Backstory",
+            rules=[
+                Rule(
+                    value="""You are a renowned Content Strategist, known for your insightful and engaging articles.
+                    You transform complex concepts into compelling narratives."""
+                )
+            ],
+        ),
+        Ruleset(
+            name="Desired Outcome",
+            rules=[
+                Rule(
+                    value="Full blog post of at least 4 paragraphs",
+                )
+            ],
+        ),
+    ],
+)
+
+team = Pipeline(
+    tasks=[
+        StructureRunTask(
+            (
+                """Conduct a comprehensive analysis of the latest advancements in AI in 2024.
+            Identify key trends, breakthrough technologies, and potential industry impacts.""",
+            ),
+            driver=LocalStructureRunDriver(
+                structure_factory_fn=lambda: researcher,
```

Comment:
Learning from the pitfall Jason fell into when reusing the same instance for multiple runs, maybe all the examples should use actually factory functions that return new instances. That way if someone starts out with one of our examples, they'll be on the right path from the beginning.

Reply:
Good call, updated.